### ISSUE TITLE
LSTMCombinedInitModel incorrectly uses DataLoader to shuffle initializer and predictor datasets.

### ISSUE THREAD

**Issue by AlexandraBaier** [AlexandraBaier/deepsysid#35]
The following lines are currently used in the train method:
```          
data_loader_init = data.DataLoader(
    initializer_dataset, self.batch_size, shuffle=True, drop_last=True
)
data_loader_pred = data.DataLoader(
    predictor_dataset, self.batch_size, shuffle=True, drop_last=True
)
```
This results in disjoint shuffling of pairs of initialization window and prediction sequence. The end of the initialization won't match with the start of the prediction sequence.

The train method is also missing calls to .zero_grad(), which might also have a negative effect on training.

**Comment by Dany-L** [comment]
True! Are you using that model for your evaluation currently? If not I would postpone it for now since the `LSTMInitModel` works quite well.

**Comment by AlexandraBaier** [comment]
I am not using it currently. Another issue with the training is that there is never any call of .zero_grad(), which is at least a bad habit to forget.
